### DISCUSSION TITLE
Using target name in index argument from slice

### DISCUSSION THREAD

**Discussion by jennysjaarda** [ropensci/targets#521] [Ideas]
I came across the `slice` feature and it's perfect for my pipeline as I slowly filter down my analyses based on previous results. However I am running into issues when I use a target name to select the indices in slice. Is this currently possible? I think it would be a great feature addition. 

Here's an example: 

```
library(targets)
library(tarchetypes)

 tar_target(random_vec, tibble(i = 1:10)),
  tar_target(random_vec2, c(1:100)),
  tar_target(random_vec2_indices, sample(random_vec2, dim(random_vec)[1])),

  tar_target(rnorm_random_vec, rnorm(random_vec$i), pattern = map(random_vec), iteration = "list"),
  tar_target(
    count_random_vec,
    {
 
      length(rnorm_random_vec) + random_vec2
    },
    pattern = map(random_vec, rnorm_random_vec, slice(random_vec2, index = random_vec2_indices))
    #map(traits_to_run, trait_info, slice(IV_data_summary, index = IV_indices_to_run)
    )


)
```

The above throws an error

```
Error : index is out of bounds in pattern = slice().
```

but if I change the index to be `1:10` for example, all works as expected.

**Comment by jennysjaarda** [comment]
I guess as a workaround we can create a new target as 
```
 tar_target(random_vec2_run, random_vec2[random_vec2_indices])

```

and use `random_vec2_run` in place of `random_vec2` in the `count_random_vec` target

**Reply by wlandau** [reply]
Yeah, in most cases I would recommend creating the subset using a different target and then branching over the subset.
